### PR TITLE
docs(Slug): add dev docs to Slug overview

### LINKED ISSUE #15784: [Feature Request]: Add developer documentation to the experimental Slug storybook
### The problem

Right now, this page just says "Hello". Let's provide some initial developer documentation for the teams currently building with the Slug and related AI-enhanced components.

https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-slug--overview

### The solution

I'm not sure what we should all document, but we should probably link to:

https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo

This Airtable so development teams can understand what's all available.

https://github.com/carbon-design-system/carbon-for-ai/blob/main/docs/support-for-carbon-v10.md

This documentation in case product teams want to use the AI-enhanced components with Carbon v10.

### Examples

_No response_

### Application/PAL

_No response_

### Business priority

None

### Available extra resources

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/15784

Adds in documentation regarding the usage of the `Slug` component and its props

#### Changelog

**New**

- Documentation about how to use the `Slug` component and a few of its props

**Changed**

- Added some doc-specific overrides for styles that were being affected. 

#### Testing / Reviewing

Ensure the spelling/grammar is correct and the content makes sense. Is there anything else we are missing?

### CODE REVIEW

**Comment by mattrosno** [comment]
@tw15egan this is excellent. What about the other components? Are there docs explaining how to pass in a Slug component into the slug prop?

**Comment by tw15egan** [comment]
@mattrosno I can add that. Would it make sense to keep it all in this overview, or should I add it to the examples folder? I’ll add some DataTable docs as well

**Comment by mattrosno** [comment]
@tw15egan at minimum, maybe something in all 4 Overviews, even if 3 of them just point you to the first that includes all documentation.

image

No strong preference here besides having something mentioned in all Overviews.

**Comment by tw15egan** [comment]
@mattrosno updated

**Review by guidari — APPROVED** [review]
LGTM! 🚀

File: packages/react/src/components/Slug/Slug-datatable.mdx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    </TableRow>
+  ))}
+</TableBody>
+```
+
+## Feedback
+
+Help us improve this component by providing feedback, asking questions on Slack,
+or updating this file on
+[GitHub](https://github.com/carbon-design-system/carbon/edit/main/packages/react/src/components/RadioButton/RadioButton.mdx).
```
Comment (mattrosno) [inline packages/react/src/components/Slug/Slug-datatable.mdx]: This link to `RadioButton` needs updated?

File: packages/react/src/components/Slug/Slug-datatable.mdx [hunk 2]
```diff
@@ -0,0 +1,153 @@
+import { ArgsTable, Canvas, Story } from '@storybook/blocks';
+import { Slug, SlugContent, SlugActions } from '../Slug';
+import * as SlugStories from '../DataTable/stories/DataTable-slug.stories';
+
+# Slug
+
+[Source code](https://github.com/carbon-design-system/carbon/tree/main/packages/react/src/components/Slug)
+|
+[Slug Release Status](https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo)
```
Comment (mattrosno) [inline packages/react/src/components/Slug/Slug-datatable.mdx]:
```suggestion
[Slug release status](https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo)
```

File: packages/react/src/components/Slug/Slug.mdx [hunk 3]
```diff
@@ -1 +1,204 @@
-Hello
+import { ArgsTable, Canvas, Story } from '@storybook/blocks';
+import { Slug, SlugContent, SlugActions } from '../Slug';
+import * as SlugStories from './Slug.stories';
+
+# Slug
+
+[Source code](https://github.com/carbon-design-system/carbon/tree/main/packages/react/src/components/Slug)
+|
+[Slug Release Status](https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo)
```
Comment (mattrosno) [inline packages/react/src/components/Slug/Slug.mdx]:
```suggestion
[Slug release status](https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo)
```

File: packages/react/src/components/Slug/SlugExamples.mdx [hunk 4]
```diff
@@ -0,0 +1,97 @@
+# Slug
+
+[Source code](https://github.com/carbon-design-system/carbon/tree/main/packages/react/src/components/Slug)
+|
+[Slug Release Status](https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo)
```
Comment (mattrosno) [inline packages/react/src/components/Slug/SlugExamples.mdx]:
```suggestion
[Slug release status](https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo)
```

File: packages/react/src/components/Slug/SlugForm.mdx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# Slug
+
+[Source code](https://github.com/carbon-design-system/carbon/tree/main/packages/react/src/components/Slug)
+|
+[Slug Release Status](https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo)
```
Comment (mattrosno) [inline packages/react/src/components/Slug/SlugForm.mdx]:
```suggestion
[Slug release status](https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo)
```

File: packages/react/src/components/Slug/SlugForm.mdx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+- [Slug API docs](https://react.carbondesignsystem.com/?path=/story/experimental-unstable-slug--overview)
+- [Slug-enabled components](https://react.carbondesignsystem.com/?path=/story/experimental-unstable-slug-examples--overview)
+- [Feedback](#feedback)
+
+## Feedback
+
+Help us improve this component by providing feedback, asking questions on Slack,
+or updating this file on
+[GitHub](https://github.com/carbon-design-system/carbon/edit/main/packages/react/src/components/RadioButton/RadioButton.mdx).
```
Comment (mattrosno) [inline packages/react/src/components/Slug/SlugForm.mdx]: Here too.

File: packages/react/src/components/Slug/SlugExamples.mdx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+);
+
+<TextInput slug={slug} />
+```
+
+## Feedback
+
+Help us improve this component by providing feedback, asking questions on Slack,
+or updating this file on
+[GitHub](https://github.com/carbon-design-system/carbon/edit/main/packages/react/src/components/RadioButton/RadioButton.mdx).
```
Comment (mattrosno) [inline packages/react/src/components/Slug/SlugExamples.mdx]: Here too.

File: packages/react/src/components/Slug/Slug.mdx [hunk 8]
```diff
@@ -1 +1,204 @@
...
+to the AI-generated state. For an example, please take a look at the
+[Revert Test story](https://react.carbondesignsystem.com/?path=/story/experimental-unstable-slug-form--revert-test)
+or take a look at the example story
+[source code](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/Slug/Slug-form.stories.js#L96).
+
+## Feedback
+
+Help us improve this component by providing feedback, asking questions on Slack,
+or updating this file on
+[GitHub](https://github.com/carbon-design-system/carbon/edit/main/packages/react/src/components/RadioButton/RadioButton.mdx).
```
Comment (mattrosno) [inline packages/react/src/components/Slug/Slug.mdx]: Here too.

**Review by mattrosno — CHANGES_REQUESTED** [review]

File: packages/react/src/components/Slug/SlugExamples.mdx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+[Slug Release Status](https://airtable.com/appCAqlGVN8tRUbAp/shr71ZyLlIGORz3Vh/tblHqPusgkK8hIeHo)
+|
+[Using AI-enhanced components in V10](https://github.com/carbon-design-system/carbon-for-ai/blob/main/docs/support-for-carbon-v10.md)
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Slug anatomy](#slug-anatomy)
+- List of components that accept `Slug`
+  - [Checkbox](https://react.carbondesignsystem.com/?path=/story/experimental-unstable-slug-examples--checkbox)
```
Comment (mattrosno) [inline packages/react/src/components/Slug/SlugExamples.mdx]: Not a big deal, but I noticed these open in new browser tabs. When we cross-link stories, do we typically use relative paths or absolute like this?

**Comment by tw15egan** [comment]
@mattrosno fixed 👍🏻

**Review by mattrosno — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**